src/data-types/datetime.ts

```diff
@@ -55,10 +66,13 @@ const DateTime: DataType = {
       buffer.writeUInt8(8);
       buffer.writeInt32LE(days);
       buffer.writeUInt32LE(threeHundredthsOfSecond);
+
+      yield buffer.buffer
```

Comment:
I think this is a typo?  🤔 `WritableTrackingBuffer.buffer` contains the whole pre-allocated buffer, which can be bigger than the data that was written in the `WritableTrackingBuffer` (and would be returned via `WritableTrackingBuffer.data`). I think that's what's causing the remaining test failures. 😬 

```suggestion
      yield buffer.data;
```

Reply:
Oh oops yeah that was a typo